.github/actions/publish/action.yml

```diff
@@ -0,0 +1,292 @@
+            CSC_LINK=${{ fromJSON(inputs.secrets).CUSTOM_JOB_SECRET_4 }}
+            ELECTRON_BUILDER_OS=win
+            TARGETS='nsis portable'
+        else
+            exit 1
+        fi
+
+        npm link electron-builder
+
+        for target in ${TARGETS}; do
+
+            # FIXME: (de)hairball this
+            ELECTRON_BUILDER_CONFIG="$(mktemp)"
+            node extend-electron-builder-config.js \
+              "$(pwd)/.resinci.json" \
+              "${APPLICATION_NAME}" > "${ELECTRON_BUILDER_CONFIG}"
+
+            PRODUCT_NAME="$(jq -r '.productName' "${ELECTRON_BUILDER_CONFIG}")"
+
+            electron-builder \
+              --config="${ELECTRON_BUILDER_CONFIG}" \
+              --c.extraMetadata.analytics.sentry.token="$(cat < '${{ runner.temp }}/sentry_dsn')" \
+              --c.extraMetadata.name="${APPLICATION_NAME}" \
+              --c.extraMetadata.packageType="${target}" \
+              --c.extraMetadata.version="${APPLICATION_VERSION}" \
+              "--${ELECTRON_BUILDER_OS}" \
+              ${ARCHITECTURE_FLAGS}
+
+            rm "${ELECTRON_BUILDER_CONFIG}"
+
+            if [[ $target =~ appimage ]]; then
+              BUILD_DIRECTORY="$(pwd)/dist"
+              APPIMAGE_PATH="${BUILD_DIRECTORY}/${PRODUCT_NAME}-${APPLICATION_VERSION}-${ELECTRON_BUILDER_ARCHITECTURE}.AppImage"
+              APPIMAGE_ZIP_PATH="${BUILD_DIRECTORY}/${APPLICATION_NAME}-${APPLICATION_VERSION}-${ELECTRON_BUILDER_OS}-${ELECTRON_BUILDER_ARCHITECTURE}.zip"
+
+              zip_file \
+                -f "${APPIMAGE_PATH}" \
+                -s "${ELECTRON_BUILDER_OS}" \
+                -o "${APPIMAGE_ZIP_PATH}"
+            fi
+        done
+
+        echo "version=${APPLICATION_VERSION}" >> $GITHUB_OUTPUT
+
+      env:
+        # Apple notarization (afterSignHook.js)
+        XCODE_APP_LOADER_EMAIL: ${{ inputs.XCODE_APP_LOADER_EMAIL }}
+        XCODE_APP_LOADER_PASSWORD: ${{ fromJSON(inputs.secrets).CUSTOM_JOB_SECRET_6 }}
+        # https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/#improvements-for-public-repository-forks
+        # https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks#about-workflow-runs-from-public-forks
+        CSC_FOR_PULL_REQUEST: true
+
+    - name: Rename release assets
+      shell: bash --noprofile --norc -eo pipefail -x {0}
+      run: |
+        set -ea
+
+        [[ '${{ inputs.VERBOSE }}' =~ on|On|Yes|yes|true|True ]] && set -x
+
+        # generalise release asset names (latest) to avoid rebuilding/renaming on finalize
+        find dist -type f -maxdepth 1 \
+          -name "*${{ steps.package_release.outputs.version }}*" \
+          -exec sh -c 'mv {} $(echo {} | sed 's/${{ steps.package_release.outputs.version }}/latest/g')' \;
+
+    # https://www.electron.build/auto-update.html#staged-rollouts
+    - name: Configure staged rollout(s)
```

Comment:
@Page- @mcraa @builder555 @dfunckt folks, sanity check please on this... This is reverse engineered `resinci-deploy editlatest ...` (from [here](https://github.com/product-os/scripts/blob/master/meta/meta-electron.sh), which never ran by the [looks](https://ci.product-os.io/teams/main/pipelines/github-events-balena-io/jobs/meta-electron) of it). Basically we now update all the electron update yaml(s) from repo.yml spec and upload to GH as releases (once!).

Reply:
well if it never ran, that explains why was `stagingPercentage` always `0` however `triggerNotification` was set